Sources/SyncDataProviders/Bookmarks/BookmarksProvider.swift

```diff
@@ -167,6 +168,13 @@ public final class BookmarksProvider: DataProvider {
             if bookmark.isPendingDeletion, !isLocalChangeRejectedBySync {
                 context.delete(bookmark)
             } else {
+                if !isLocalChangeRejectedBySync, bookmark.isFolder {
```

Comment:
This is inside `cleanUpSentItems` which is called before processing the response. If a bookmark was rejected by sync, it comes with the response, will be processed there and its sync payload will be updated then.

Reply:
`isPendingDeletion` is required to handle "soft deletion". Objects that are deleted from the UI aren't removed from the database, but instead are marked as pending deletion and filtered out in the UI. They are only deleted after the deletion is synced to the server.